packages/react/src/FormControl/FormControl.tsx

```diff
@@ -99,9 +100,9 @@ const FormControl = React.forwardRef<HTMLDivElement, FormControlProps>(
         )
       }
 
-      if (childrenWithoutSlots.find(child => React.isValidElement(child) && child.props?.required)) {
+      if (isRadioInput && childrenWithoutSlots.find(child => React.isValidElement(child) && child.props?.required)) {
```

Comment:
Do we also cover the case of ["Radio groups can only have the group receive a required setting because you can only select one choice."](https://github.com/github/primer/issues/3470#:~:text=Radio%20groups%20can%20only%20have%20the%20group%20receive%20a%20required%20setting%20because%20you%20can%20only%20select%20one%20choice.)

Sorry if I missed it!

Reply:
Yeah so that'd be handled at the [RadioGroup](https://primer.style/react/storybook/?path=/story/components-radiogroup--playground&args=required:!true&globals=colorScheme:light) level. The FormControl will be contained within the RadioGroup in this case (RadioGroup -> FormControl -> Radio) and doesn't need to be concerned about required being on the parent. Radios will not receive `required` from the FromControl regardless

Reply:
Sounds good! Thanks for clarifying 🙌🏻